### PR TITLE
Run precommit in its own job, cache the data

### PR DESCRIPTION
**Pull request by chriselion**
* Move pre-commit to it's own job. Currently it runs on all 3 python testing jobs.
* Cache the pre-commit cache (and venv). Timings below.

Should we also make the python unit test jobs dependent on precommit job?

Resolves:
https://jira.unity3d.com/browse/MLA-17
https://jira.unity3d.com/browse/MLA-394

### CODE REVIEW

**Comment by chriselion** [comment]
Cache miss:
image

Cache hit (precommit only)
image

Cache hit (precommit + venv)
image

Total time for the run goes from 1:44 uncached to 0:42 cached.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -55,14 +55,14 @@ repos:
         types: [markdown]
     -   id: check-merge-conflict
         args: [--assume-in-merge]
+    -   id: check-yaml
```
Comment (chriselion) [inline .pre-commit-config.yaml]: I can back this out, but I it saved me at least one bad `config.yml` push.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -55,14 +55,14 @@ repos:
...
+        exclude: \.yamato/*
```
Comment (surfnerd) [inline .pre-commit-config.yaml]: were the Yamato configs failing?

Comment (chriselion) [reply .pre-commit-config.yaml]: (discussed offline, templating breaks the checks)

Comment (chriselion) [reply .pre-commit-config.yaml]: (discussed offline, templating breaks the checks)

**Pull request merged**